### PR TITLE
assign provider on pre-connected layer one card

### PR DESCRIPTION
**Pull request by Aierie**
CS-653

- Adds `currentProviderId` property to the `Layer1WebStrategy` type
- Assigns provider for layer one connection card by checking `layer1Network.strategy.currentProviderId` if there is a connected layer 1 account.

### CODE REVIEW

**Review by lukemelia — CHANGES_REQUESTED** [review]
Let's add a test.

File: packages/web-client/app/components/card-pay/layer-one-connect-card/index.ts [hunk 1]
```diff
@@ -36,24 +36,23 @@ class CardPayDepositWorkflowConnectLayer1Component extends Component<CardPayDepo
...
     if (this.hasAccount) {
       next(this, () => {
-        this.currentWalletProviderId =
-          this.layer1Network.strategy.currentProviderId || '';
         this.args.onComplete?.();
       });
     }
   }
-  get currentWalletProvider(): WalletProvider | undefined {
+  get connectedWalletProvider(): WalletProvider | undefined {
```
Comment (Aierie) [inline packages/web-client/app/components/card-pay/layer-one-connect-card/index.ts:43]: When writing the test for this I realised that the `LayerOneConnectCard` component  could get its provider from the network's strategy directly, so it's now separated from the selection state.

Comment (lukemelia) [reply packages/web-client/app/components/card-pay/layer-one-connect-card/index.ts:43]: Tangential benefit of writing tests!

**Review by lukemelia — APPROVED** [review]
:shipit: !

**Pull request merged**